models/GroupsService.php

```diff
@@ -74,48 +95,47 @@ public function deleteGroup(\core_kernel_classes_Resource $group)
      *
      * @access public
      * @author Joel Bout, <joel.bout@tudor.lu>
-     * @param  Class clazz
+     * @param  core_kernel_classes_Class $clazz
      * @return boolean
      */
     public function isGroupClass(core_kernel_classes_Class $clazz)
     {
-        return $clazz->equals($this->getRootClass()) || $clazz->isSubClassOf($this->getRootClass());
+        return $clazz->equals($this->getRootClass())
+            || $clazz->isSubClassOf($this->getRootClass());
     }
 
     /**
-     * get the groups of a user
+     * Get the groups of a user
      *
      * @access public
      * @author Joel Bout, <joel.bout@tudor.lu>
-     * @param  string userUri
+     * @param  User $user
      * @return array resources of group
      */
     public function getGroups(User $user)
     {
-        $groups = $user->getPropertyValues(self::PROPERTY_MEMBERS_URI);
-        array_walk($groups, function (&$group) {
-            $group = new core_kernel_classes_Resource($group);
-        });
-        return $groups;
+        return array_map(
+            function (string $group): core_kernel_classes_Resource {
+                return $this->getModel()->getResource($group);
+            },
+            $user->getPropertyValues(self::PROPERTY_MEMBERS_URI)
+        );
```

Comment:
```suggestion
        return array_map(
            fn (string $group): core_kernel_classes_Resource => $this->getModel()->getResource($group),
            $user->getPropertyValues(self::PROPERTY_MEMBERS_URI)
        );
```

Reply:
I always forgot arrow functions are available in PHP 7.4 :sweat_smile: Applying this suggestion, thanks :slightly_smiling_face: